src/lua/functions/creatures/monster/monster_type_functions.cpp

```diff
@@ -676,55 +679,53 @@ int MonsterTypeFunctions::luaMonsterTypeConditionImmunities(lua_State* L) {
 	if (monsterType) {
 		if (lua_gettop(L) == 1) {
-			lua_pushnumber(L, monsterType->info.conditionImmunities);
+			lua_createtable(L, monsterType->info.conditionImmunities.size(), 0);
+			int8_t conditionCount = -1;
+			for (auto &condition : monsterType->info.conditionImmunities) {
+				conditionCount++;
+				lua_pushnumber(L, condition);
+				lua_rawseti(L, -2, conditionCount);
+			}
 		} else {
 			std::string immunity = getString(L, 2);
+			ConditionType_t condition = CONDITION_NONE;
 			if (immunity == "physical") {
-				monsterType->info.conditionImmunities |= CONDITION_BLEEDING;
-				pushBoolean(L, true);
+				condition = CONDITION_BLEEDING;
 			} else if (immunity == "energy") {
-				monsterType->info.conditionImmunities |= CONDITION_ENERGY;
-				pushBoolean(L, true);
+				condition = CONDITION_ENERGY;
 			} else if (immunity == "fire") {
-				monsterType->info.conditionImmunities |= CONDITION_FIRE;
-				pushBoolean(L, true);
+				condition = CONDITION_FIRE;
 			} else if (immunity == "poison" || immunity == "earth") {
-				monsterType->info.conditionImmunities |= CONDITION_POISON;
-				pushBoolean(L, true);
+				condition = CONDITION_POISON;
 			} else if (immunity == "drown") {
-				monsterType->info.conditionImmunities |= CONDITION_DROWN;
-				pushBoolean(L, true);
+				condition = CONDITION_DROWN;
 			} else if (immunity == "ice") {
-				monsterType->info.conditionImmunities |= CONDITION_FREEZING;
-				pushBoolean(L, true);
+				condition = CONDITION_FREEZING;
 			} else if (immunity == "holy") {
-				monsterType->info.conditionImmunities |= CONDITION_DAZZLED;
-				pushBoolean(L, true);
+				condition = CONDITION_DAZZLED;
 			} else if (immunity == "death") {
-				monsterType->info.conditionImmunities |= CONDITION_CURSED;
-				pushBoolean(L, true);
+				condition = CONDITION_CURSED;
 			} else if (immunity == "paralyze") {
-				monsterType->info.conditionImmunities |= CONDITION_PARALYZE;
-				pushBoolean(L, true);
+				condition = CONDITION_PARALYZE;
 			} else if (immunity == "outfit") {
-				monsterType->info.conditionImmunities |= CONDITION_OUTFIT;
-				pushBoolean(L, true);
+				condition = CONDITION_OUTFIT;
 			} else if (immunity == "drunk") {
-				monsterType->info.conditionImmunities |= CONDITION_DRUNK;
-				pushBoolean(L, true);
+				condition = CONDITION_DRUNK;
 			} else if (immunity == "invisible" || immunity == "invisibility") {
-				monsterType->info.conditionImmunities |= CONDITION_INVISIBLE;
-				pushBoolean(L, true);
+				condition = CONDITION_INVISIBLE;
 			} else if (immunity == "bleed") {
-				monsterType->info.conditionImmunities |= CONDITION_BLEEDING;
-				pushBoolean(L, true);
+				condition = CONDITION_BLEEDING;
 			} else {
 				SPDLOG_WARN("[MonsterTypeFunctions::luaMonsterTypeConditionImmunities] - "
 							"Unknown immunity name: {} for monster: {}",
 							immunity, monsterType->name);
 				lua_pushnil(L);
 			}
+
+			monsterType->info.conditionImmunities[condition] = CONDITION_NONE;
```

Comment:
```suggestion
			monsterType->info.conditionImmunities[condition] = condition;
```
We check for type == type in `Creature::isImmune` so we need this to match